src/main/java/aquality/selenium/browser/IBrowserTabNavigation.java

```diff
@@ -87,10 +88,38 @@ default void openNewTab() {
      */
     void openNewTab(boolean switchToNew);
 
+    /**
+     * Opens and switches to new tab using JS function.
+     */
+    default void openNewTabViaJs() {
+        openNewTabViaJs(true);
+    }
+
+    /**
+     * Opens new tab using JS function.
+     *
+     * @param switchToNew Switches to new tab if true and stays at current otherwise.
+     */
+    void openNewTabViaJs(boolean switchToNew);
+
     /**
      * Navigates to desired url in new tab.
      *
      * @param url String representation of URL.
      */
     void openInNewTab(String url);
+
+    /**
+     * Navigates to desired url in new tab.
+     *
+     * @param url target URL.
+     */
+    void openInNewTab(URL url);
+
+    /**
+     * Navigates to desired url in new tab.
+     *
+     * @param url String representation of URL.
+     */
+    void openInNewTabViaJs(String url);
```

Comment:
Don't we use something pattern like:
browser.GetJsActions().DoSoemthingWithJS() ?
If so, should not we follow this?
I understand that this is not simple question in this situation, but ages ago we voted to introduce these JsActions container instead of having things like ClickJs and etc.

Maybe, we need to have interface like
browser.GetJsActions().OpenNewTab()?

Reply:
No we don't :) There are no such thing as BrowserJsActions in our framework. Browser is jsExecutor and can execute any js; although, we have an enum with predefined scripts listed. 
There are Element/JsActions, but that's a different story.
I have created openInNewTabViaJs method to preserve previous behavior in case somebody needs it. It is still related to "TabNavigation" purpose, so can live in the same interface. If we extract everything we do via JS into some BrowserJsActions interface we might break single responsibility principle, don't you think?